test/test-document-c14n.lua

```diff
@@ -0,0 +1,73 @@
+local luaunit = require("luaunit")
+local xmlua = require("xmlua")
+local ffi = require("ffi")
+
+TestDocumentC14N = {}
+
+-- From: https://www.w3.org/2008/xmlsec/Drafts/c14n-20/test-cases/Overview.src.html
+
+function TestDocumentC14N.test_PIs_Comments_and_Outside_of_Document_Element()
+  -- https://www.w3.org/2008/xmlsec/Drafts/c14n-20/test-cases/files/inC14N1.xml
```

Comment:
Ah, we don't need to test C14N behaviors implemented in libxml2. Because they must be tested in libxml2.

We only need to test our API. For example, `document:canonicalize(function)` case, `document:canonicalize(nodes)` case, document:canonicalize(..., options)` case and so on.

Reply:
agreed, all I did was just grab some of them to verify it works as intended.
This tests with and without "with_comments" probably also needs a test with the prefix list